packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm

```diff
@@ -261,9 +261,10 @@ - (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &
   }
 
   // `transform`
-  if (oldViewProps.transform != newViewProps.transform &&
-      ![_propKeysManagedByAnimated_DO_NOT_USE_THIS_IS_BROKEN containsObject:@"transform"]) {
-    self.layer.transform = RCTCATransform3DFromTransformMatrix(newViewProps.transform);
+  if ((oldViewProps.transform != newViewProps.transform || oldViewProps.transformOrigin != newViewProps.transformOrigin)
+      && ![_propKeysManagedByAnimated_DO_NOT_USE_THIS_IS_BROKEN containsObject:@"transform"]) {
+    auto newTransform = newViewProps.resolveTransform(_layoutMetrics);
+    self.layer.transform = RCTCATransform3DFromTransformMatrix(newTransform);
```

Comment:
We should move this logic to `invalidateLayer`, then in here just set `needsInvalidateLayer = YES`. Reason being (I believe) `invalidateLayer` is called when the view is resized, but `updateProps` is not so you can't use `_layoutMetrics` there

Reply:
Good catch! it also removed a redundant call in updateLayoutMetrics. https://github.com/facebook/react-native/pull/38559/commits/26b8430ef929af1fae2a447e65a77d91512fca9d

Reply:
Thinking about it again. Each transform animation can cause the layer to invalidate and it can trigger a lot of invalidateLayer calls. But yeah this same happens with opacity now. So not sure if it is good!

Reply:
Yeah that's true - maybe pull it out into a `updateTransform` function that you call on layer invalidation and when the transform changes